// react-progressbar/index.js
/**
 * @jsx React.DOM
 */
'use strict';

var React = require('react');

var component = React.createClass({
  propTypes: {
    onProgressChange: React.PropTypes.func,
  },

  render: function() {
            
    var completed = +this.props.completed;
    if (completed < 0) {completed = 0};
    if (completed > 100) {completed = 100};

    var style = {
      backgroundColor: this.props.color || '#0BD318',
      width: completed + '%',
      transition: "width 200ms",
      height: 10
    };

    return (
      <div ref='parent' className="progressbar-container" onClick = {this._handleClick}>
        <div className="progressbar-progress" style={style}>
        </div>
      </div>
    );
  },

   _handleClick: function(evt, args) {
	var div = this.refs.parent;
	if (!div) {
	    return;
	}
	
	var rect = div.getDOMNode().getBoundingClientRect();
	var left = rect.left; 
	var width = rect.width;
	var click_left = evt.pageX;

	var percent = (click_left - left) / width * 100;
	if (this.props.onProgressChange) {
	    this.props.onProgressChange(percent);
	}
    }
});

module.exports = component;
